HandleNHDeleteInstallationCallOrchestrator/handler.ts

```diff
@@ -24,51 +32,143 @@ export type NhDeleteInstallationOrchestratorCallInput = t.TypeOf<
+  const verbose: string =
+    failure.kind === "FAILURE_INVALID_INPUT"
+      ? `ERROR=${failure.reason}|INPUT=${toString(failure.input)}`
+      : failure.kind === "FAILURE_ACTIVITY"
+      ? `ERROR=${failure.reason}|ACTIVITY=${failure.activityName}`
+      : failure.kind === "FAILURE_UNHANDLED"
+      ? `ERROR=${failure.reason}`
+      : defaultNever(
+          failure,
+          `unknown failure kind, failure: ${toString(failure)}`
+        );
+  context.log.error(log);
+  context.log.verbose(`${log}|${verbose}`);
 };
 
+function* deleteInstallation(
+  context: IOrchestrationFunctionContext,
+  input: NHDeleteInstallationActivityInput,
+  retryOptions: df.RetryOptions
+): Generator<Task, "SUCCESS"> {
+  const activityName = "HandleNHDeleteInstallationCallActivity";
+  const result = yield context.df.callActivityWithRetry(
+    activityName,
+    retryOptions,
+    input
+  );
+
+  return either
+    .of<ActivityResultFailure | Error, unknown>(result)
+    .chain(r =>
+      ActivityResult.decode(r).mapLeft(
+        e =>
+          new Error(
+            `Cannot decode result from ${activityName}, err: ${readableReport(
+              e
+            )}`
+          )
+      )
+    )
+    .chain(r => (ActivityResultSuccess.is(r) ? right(r) : left(r)))
+    .fold(
+      // In case of failure, trow a failure object with the activity name
+      e => {
+        throw OrchestratorActivityFailure.encode({
+          activityName,
+          kind: "FAILURE_ACTIVITY",
+          reason: e instanceof Error ? e.message : e.reason
+        });
+      },
+      _ => "SUCCESS" as const
+    );
+}
+
 export const getHandler = (config: IConfig) =>
   function*(context: IOrchestrationFunctionContext): Generator<unknown> {
-    const logPrefix = `NhDeleteInstallationOrchestratorCallInput`;
-
     const retryOptions = {
       ...new df.RetryOptions(5000, config.RETRY_ATTEMPT_NUMBER),
       backoffCoefficient: 1.5
     };
 
-    // Get and decode orchestrator input
-    const input = context.df.getInput();
-    const errorOrNHCallOrchestratorInput = NhDeleteInstallationOrchestratorCallInput.decode(
-      input
-    );
-
-    if (isLeft(errorOrNHCallOrchestratorInput)) {
-      logError(context, logPrefix, errorOrNHCallOrchestratorInput);
-      return false;
+    try {
+      // Decode delete message from input
+      const input = context.df.getInput();
+      const { message } = NhDeleteInstallationOrchestratorCallInput.decode(
+        input
+      ).getOrElseL(err => {
+        throw OrchestratorInvalidInputFailure.encode({
+          input,
+          kind: "FAILURE_INVALID_INPUT",
+          reason: `${readableReport(err)}`
+        });
+      });
+
+      yield* deleteInstallation(
+        context,
+        {
+          installationId: message.installationId,
+          notificationHubConfig: getNHLegacyConfig(config)
+        },
+        retryOptions
+      );
+
+      return OrchestratorSuccess.encode({ kind: "SUCCESS" });
```

Comment:
Why the need of this `OrchestratorSuccess`? I mean, we can find in logs?

Reply:
It is logged in the taskhub table storage, in the `ExecutionCompleted` event. 
Btw here it's added only for consistency as it does not add any further info